brave/brave-core
# Add more user-friendly messaging for tip dialog errors

Resolves https://github.com/brave/brave-browser/issues/12224

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

Designs can be found in the "error states" section [here](https://www.figma.com/file/EDGI52tfWG1ggC9ZYFguXy/tipping-banner-2.0?node-id=2595%3A0).

In addition to the STR described in the issue, the UX for tip dialog errors can simulated with the following procedure:

- Open the browser, pointed to rewards staging
- Navigate to a valid publisher, such as https://laurenwags.github.io/
- Click the rewards panel and click "Send a tip"
- Right-click in the tip dialog, and select "Inspect"
- In the devtools window, select the "Console" tab.
- Execute the following code:
  - `window[Symbol.for('setTipDialogErrorForTesting')]('ERR_INVALID_DIALOG_ARGS')`
  - Verify that a generic error message is displayed matching designs
- Execute the following code:
  - `window[Symbol.for('setTipDialogErrorForTesting')]('ERR_FETCH_BALANCE', 1)`
  - Verify that a "server connection" error message is displayed matching designs.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
